lldb/source/Commands/CommandObjectProcess.cpp

```diff
@@ -1304,9 +1304,11 @@ class CommandObjectProcessSaveCore : public CommandObjectParsed {
         FileSpec output_file(command.GetArgumentAtIndex(0));
         FileSystem::Instance().Resolve(output_file);
         SaveCoreStyle corefile_style = m_options.m_requested_save_core_style;
-        Status error =
-            PluginManager::SaveCore(process_sp, output_file, corefile_style,
-                                    m_options.m_requested_plugin_name);
+        CoreDumpOptions core_dump_options(output_file);
+        core_dump_options.SetCoreDumpPluginName(
+            m_options.m_requested_plugin_name);
+        core_dump_options.SetCoreDumpStyle(corefile_style);
+        Status error = PluginManager::SaveCore(process_sp, core_dump_options);
```

Comment:
We need to remove the instance variables `m_requested_plugin_name` and `m_requested_save_core_style` and replace with with `lldb_private::CoreDumpOptions m_options;`. Then in the code that used to fill in the old ivars, we will just call the accessor instead. We can also call the accessor to set the output file as well on the `m_options;` ivar.